app/models/miq_set.rb

```diff
@@ -0,0 +1,66 @@
+class MiqSet < ApplicationRecord
+  # self.abstract_class = true
+  self.inheritance_column = :set_type
+
+  include UuidMixin
+
+  acts_as_miq_taggable
+
+  serialize :set_data
+
+  validates :name, :presence => true, :uniqueness_when_changed => {:scope => [:set_type, :userid, :group_id], :if => proc { |c| c.class.in_my_region.exists?(:name => c.name) }}
+  validates :description, :presence => true
+
+  has_many :miq_set_memberships, :dependent => :delete_all
+  belongs_to :owner, :polymorphic => true
+
+  attr_writer :model_class
+
+  def self.model_class
```

Comment:
almost wish we just declared this instead of figuring it out
(like rails does with table_name)

hardcoding the name and string munging is sketchy

I'd run this by @Fryguy before making this change though

Reply:
Hey @Fryguy could I get a comment from you on this one, please?

Reply:
@Fryguy ping

Reply:
If there's a naming convention I generally prefer automatically figuring it out; convention over configuration... One less thing for the developer to have to remember to explicitly add.